src/output.rs

```diff
@@ -1,693 +1,674 @@
-    ///
-    /// Each name is unique among all wl_output globals, but if a wl_output
-    /// global is destroyed the same name may be reused later. The names will
-    /// also remain consistent across sessions with the same hardware and
-    /// software configuration.
-    ///
-    /// Examples of names include 'HDMI-A-1', 'WL-1', 'X11-1', etc. However, do
-    /// not assume that the name is a reflection of an underlying DRM connector,
-    /// X11 connection, etc.
-    ///
-    /// Note that this is not filled in by version 3 of the wl_output protocol,
-    /// but it has been proposed for inclusion in version 4.  Until then, it is
-    /// only filled in if your environment has an [XdgOutputHandler] global
-    /// handler for [ZxdgOutputManagerV1].
-    pub name: String,
-    /// The description of this output as advertised by the server
-    ///
-    /// The description is a UTF-8 string with no convention defined for its
-    /// contents. The description is not guaranteed to be unique among all
-    /// wl_output globals. Examples might include 'Foocorp 11" Display' or
-    /// 'Virtual X11 output via :1'.
-    ///
-    /// Note that this is not filled in by version 3 of the wl_output protocol,
-    /// but it has been proposed for inclusion in version 4.  Until then, it is
-    /// only filled in if your environment has an [XdgOutputHandler] global
-    /// handler for [ZxdgOutputManagerV1].
-    pub description: String,
-    /// Location of the top-left corner of this output in compositor
-    /// space
-    ///
-    /// Note that the compositor may decide to always report (0,0) if
-    /// it decides clients are not allowed to know this information.
-    pub location: (i32, i32),
-    /// Physical dimensions of this output, in unspecified units
-    pub physical_size: (i32, i32),
-    /// The subpixel layout for this output
-    pub subpixel: Subpixel,
-    /// The current transformation applied to this output
-    ///
-    /// You can pre-render your buffers taking this information
-    /// into account and advertising it via `wl_buffer.set_tranform`
-    /// for better performances.
-    pub transform: Transform,
-    /// The scaling factor of this output
-    ///
-    /// Any buffer whose scaling factor does not match the one
-    /// of the output it is displayed on will be rescaled accordingly.
-    ///
-    /// For example, a buffer of scaling factor 1 will be doubled in
-    /// size if the output scaling factor is 2.
-    pub scale_factor: i32,
-    /// Possible modes for an output
-    pub modes: Vec<Mode>,
-    /// Has this output been unadvertized by the registry
-    ///
-    /// If this is the case, it has become inert, you might want to
-    /// call its `release()` method if you don't plan to use it any
-    /// longer.
-    pub obsolete: bool,
+#[derive(Debug)]
+pub struct OutputState {
+    xdg: Option<ZxdgOutputManagerV1>,
+    outputs: Vec<OutputInner>,
 }
```

Comment:
The `OutputState` should also provide a method that allows iterating over the outputs. And I think we would be better of with a freestanding function that allows to retrieve the `OutputInfo` from a `WlOutput`, and just expose the `WlOutput` to the user and let them get the info using that method if they need it.